docs/source/user_guide.rst

```diff
@@ -0,0 +1,606 @@
+the cypher retrievers. This function takes the neo4j record returned by the retrieval query
+and must return a `RetrieverResultItem` with content (str) and metadata (dict) fields.
+The content is the one that will be passed to the LLM, metadata can be used for debugging
+purposes for instance.
+
+
+.. code:: python
+
+    def result_formatter(record: neo4j.Record) -> RetrieverResultItem:
+        return RetrieverResultItem(
+            content=f"Movie title: {record.get('movieTitle')}, description: {record.get('movieDescription')}, actors: {record.get('actors')}",
+            metadata={
+                "title": record.get('movieTitle'),
+                "score": record.get("score"),
+            }
+        )
+
+    retriever = VectorCypherRetriever(
+        driver,
+        index_name=INDEX_NAME,
+        retrieval_query="MATCH (node)<-[:ACTED_IN]-(p:Person) RETURN node.title as movieTitle, node.plot as movieDescription, collect(p.name) as actors, score",
+        result_formatter=result_formatter,
+    )
+
+Also see :ref:`vectorcypherretriever`.
+
+Vector Databases
+====================
+
+.. note::
+
+    For external retrievers, the filter syntax depends on the provider. Please refer to
+    the documentation of the Python client for each provider for details.
+
+
+Weaviate retrievers
+-------------------
+
+.. note::
+
+    In order to import this retriever, you must install the Weaviate Python client:
+    `pip install weaviate-client`
+
+
+.. code:: python
+
+    from weaviate.connect.helpers import connect_to_local
+    from neo4j_genai.retrievers import WeaviateNeo4jRetriever
+
+    client = connect_to_local()
+    retriever = WeaviateNeo4jRetriever(
+        driver=driver,
+        client=client,
+        embedder=embeder,
+        collection="Movies",
+        id_property_external="neo4j_id",
+        id_property_neo4j="id",
+    )
+
+Internally, this retriever performs the vector search in Weaviate, finds the corresponding node by matching
+the Weaviate metadata `id_property_external` with a Neo4j `node.id_property_neo4j`, and returns the matched node.
+
+Similarly to the vector retriever, you can also use `return_properties` or `retrieval_query` parameters.
+
+Also see :ref:`weaviateneo4jretriever`.
+
+Pinecone retrievers
+-------------------
+
+.. note::
+
+    In order to import this retriever, you must install the Weaviate Python client:
+    `pip install pinecone-client`
+
+
+.. code:: python
+
+    from pinecone import Pinecone
+    from neo4j_genai.retrievers import PineconeNeo4jRetriever
+
+    client = Pinecone()  # ... create your Pinecone client
+
+    retriever = PineconeNeo4jRetriever(
+        driver=driver,
+        client=client,
+        index_name="Movies",
+        id_property_neo4j="id",
+        embedder=embeder,
+    )
+
+Also see :ref:`pineconeneo4jretriever`.
+
+
+Other retrievers
+===================
+
+Hybrid and Hybrid Cypher retrievers
```

Comment:
Will you add anything to this section? If not perhaps just add what is the difference between vector retrievers i.e. you can search over a fulltext index as well

Reply:
Added some content, thanks.